com.unity.netcode.gameobjects/Prototyping/NetworkTransform.cs

```diff
@@ -21,106 +24,159 @@ internal struct NetworkState : INetworkSerializable
             {
                 get => (Bitset & (1 << RotAngleXBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleXBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << RotAngleXBit);
             }
+
             public bool HasRotAngleY
             {
                 get => (Bitset & (1 << RotAngleYBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleYBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << RotAngleYBit);
             }
+
             public bool HasRotAngleZ
             {
                 get => (Bitset & (1 << RotAngleZBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << RotAngleZBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << RotAngleZBit);
             }
+
             // Scale
             public bool HasScaleX
             {
                 get => (Bitset & (1 << ScaleXBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleXBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << ScaleXBit);
             }
+
             public bool HasScaleY
             {
                 get => (Bitset & (1 << ScaleYBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleYBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << ScaleYBit);
             }
+
             public bool HasScaleZ
             {
                 get => (Bitset & (1 << ScaleZBit)) != 0;
-                set => Bitset |= (ushort)((value ? 1 : 0) << ScaleZBit);
+                set => Bitset |= (ushort) ((value ? 1 : 0) << ScaleZBit);
             }
 
             public float PositionX, PositionY, PositionZ;
             public float RotAngleX, RotAngleY, RotAngleZ;
             public float ScaleX, ScaleY, ScaleZ;
+            public double SentTime;
+
+            public Vector3 Position
+            {
+                get { return new Vector3(PositionX, PositionY, PositionZ); }
+                set
+                {
+                    PositionX = value.x;
+                    PositionY = value.y;
+                    PositionZ = value.z;
+                }
+            }
+
+            public Vector3 Rotation
+            {
+                get { return new Vector3(RotAngleX, RotAngleY, RotAngleZ); }
+                set
+                {
+                    RotAngleX = value.x;
+                    RotAngleY = value.y;
+                    RotAngleZ = value.z;
+                }
+            }
+
+            public Vector3 Scale
+            {
+                get { return new Vector3(ScaleX, ScaleY, ScaleZ); }
+                set
+                {
+                    ScaleX = value.x;
+                    ScaleY = value.y;
+                    ScaleZ = value.z;
+                }
+            }
 
             public void NetworkSerialize(NetworkSerializer serializer)
             {
+                serializer.Serialize(ref SentTime);
```

Comment:
are we "always" going to send 64-bits `SentTime` with the state?
also shouldn't we put this at the bottom to match property ordering?
and does it have to be 64-bits or can we get away with 32-bits float?
and finally, why suddenly those new empty lines in this class? :)

Reply:
1. "Always" --> yes, this is needed to know when to consume from the buffer any values coming from the state.
2. I might be unaware of perf advantages for ordering this. I put this before Bitset, since it's not controlled by it (it felt more logical to have non-bitset stuff + bitset + bitset stuff, instead of mixing bitset stuff and non-bitset stuff)
3. We're comparing to NetworkTime's time double. The same reason NetworkTime uses double would apply here (@LukeStampfli if I remember correctly, your RFC mentioned being able to support long running games?). We're not sending an int tick number either since users might want to update transform on FixedUpdate, not on tick so you'd run into aliasing issues.

We could delta compress this, but it might start to become out of scope for this PR?

Reply:
3. Is correct. If we want to support dedicated servers with long running sessions (5 hours +) we need double precision.

Reply:
bump

Reply:
So I think for now all we can do is just send the 64 bits. With the snapshot system I hope we will have a smarter way to not include this time in every transform / networkvariable we send but instead put it into a meta header which we put inside the snapshot just once.